core/base/triangulation/Triangulation.h

```diff
@@ -3100,18 +3102,40 @@ namespace ttk {
                             const SimplexId &yDim,
                             const SimplexId &zDim) {
 
-      abstractTriangulation_ = &implicitTriangulation_;
-
       gridDimensions_[0] = xDim;
       gridDimensions_[1] = yDim;
       gridDimensions_[2] = zDim;
 
-      return implicitTriangulation_.setInputGrid(xOrigin, yOrigin, zOrigin,
-                                                 xSpacing, ySpacing, zSpacing,
-                                                 xDim, yDim, zDim);
+      int ret = periodicImplicitTriangulation_.setInputGrid(
+        xOrigin, yOrigin, zOrigin, xSpacing, ySpacing, zSpacing, xDim, yDim,
+        zDim);
+      int retPeriodic = implicitTriangulation_.setInputGrid(
```

Comment:
it seems that the variables "ret" and "retPeriodic" need to be switched.

Reply:
Corrected in latest commit.